core/assertion.ts

```diff
@@ -86,13 +92,14 @@ export class AssertionContext {
     this.assertion = assertion;
   }
 
-  public ref(name: string) {
+  public ref(ref: Resolvable) {
+    const name = typeof ref === "object" ? ref.schema + "." + ref.name : ref;
```

Comment:
nit: invert this and use `typeof ref === "string"? (in a few places)

Reply:
There is an actual reason why I compare it this way. The default case should include `undefined` so that you can do a `ref(undefined)`:

https://github.com/dataform-co/dataform/blob/c34588e0dd0057f1767b6d7517cae95a2656e379/tests/core/core.spec.ts#L382

But I see your point in doing this in a more type-safe fashion. I will break down the `Resolvable` part (the non-string one) into a different type, like this:

```
export type Resolvable = string  | {  schema: string;  name: string; };
```

into

````
export type FullyQualifiedName = {schema: string, name:string};
export type Resolvable = string | FullyQualifiedName;
````